plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend

```diff
@@ -378,6 +379,40 @@ class ASTStructureValidator {
 			constraints
 		)
 	}
+	
+	def private dispatch void validateASTStructure(
+		CoalesceExpression model,
+		ASTStructureDiagnosticProducer producer,
+		Set<LabelledStatement> validLabels,
+		Constraints constraints
+	) {
+		val container = model.eContainer 
+		if (container instanceof BinaryLogicalExpression) {
+			val target = NodeModelUtils.findActualNodeFor(model)
+			producer.node = target
+			producer.addDiagnostic(
+					new DiagnosticMessage(IssueCodes.getMessageForAST_INVALID_COALESCE_PARENT(container.op.literal),
+						IssueCodes.getDefaultSeverity(IssueCodes.AST_INVALID_COALESCE_PARENT), IssueCodes.AST_INVALID_COALESCE_PARENT))
+		} else if (model.expression instanceof BinaryLogicalExpression) {
```

Comment:
I understand that this is checkin in test case `testCoalesce_03()` using `x||y ?? zonk`. What about parentheses like in `(x||y) ?? zonk`?

Reply:
My understanding from the spec is, that `(x||y) ?? zonk` is allowed

Reply:
I see; parentheses heal it.

Reply:
Done